google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -1327,4 +1329,54 @@ public void floatIncrement() throws ExecutionException, InterruptedException {
       this.error = error;
     }
   }
+
+  @Test
+  public void getAllWithObserver() throws Exception {
+    DocumentReference ref = randomColl.document("doc1");
+    ref.set(ALL_SUPPORTED_TYPES_MAP).get();
+
+    DocumentReference ref1 = randomColl.document("doc2");
+    ref1.set(ALL_SUPPORTED_TYPES_MAP).get();
+
+    DocumentReference ref2 = randomColl.document("doc3");
+    ref2.set(ALL_SUPPORTED_TYPES_MAP).get();
+    ref2.delete();
+
+    final List<DocumentSnapshot> documentSnapshots = new ArrayList<>();
```

Comment:
This should be wrapped in a synchronized list since it is being written and read from different threads.

```suggestion
    final List<DocumentSnapshot> documentSnapshots = Collections.synchronizedList(new ArrayList<DocumentSnapshot>());
```

Reply:
done